### DISCUSSION TITLE
Migration with added root category + SEO

### DISCUSSION THREAD

**Discussion by jdupont22** [PrestaShop/PrestaShop#33727] [Q&A] [answered]
Hi,

I'm currently migrating from Prestashop 1.3 to the last version, 8.X. 
I'm a beginner in Prestashop, but I'm a software developer, with a very good background in SQL. 

What I understand from the default data in "ps_category" (fresh install), the version 8.X must have some default categories (I don't know since what version > ?). 

(id=1, parent_id=0)  -> Root Category (with is_root_category=0)
(id=2, parent_id=1)  -> Home Category (with is_root_category=1)

The old prestashop has a category tree without the "Root Category". 
So during my SQL migration I've create the root category with ID=1 and changing old IDs to ID=ID+1.
 
My problem is related to the SEO, all google reference are link to something like /25-category-name, but now with this +1 in the ID, my new link is /26-category-name. 

Maybe I'm missing something for my SQL migration ? Is there nya way to keep the old IDs ? 

I've try to trick the URL Rewrite, with something like '25-category-name' and changing the category format in the SEO page from {id}-{rewrite} to {rewrite} but the {id} is mandatory, too bad :D

-- Edit --

Instead of having "Root Category" with ID=1, I put the new "Root Category" to the last ID in table ps_category, like for exemple (id=281, parent_id=0).
Next,  I have link category with ID=1 (the old "Home") to the parent_id=281.
Next, I have change the "ps_configuration" table with
 - (name="PS_ROOT_CATEGORY", value=281)
 - (name="PS_HOME_CATEGORY", value=1)

Almost everything work great, except when I try to add a new category, the parent category field always propose a tree from category_id=2! I would like choosing from category_id=1 of course, the one set in "PS_HOME_CATEGORY"

Someone has an idea? 

-- Edit 2 --

After 1 hour reading / searching in the PHP code (I'm a JAVA + Typescript developer), I found the "why"!!!!
The default 'id_category' is set under the table "ps_shop", pffff, what a day! Default was st to "2" need to change to "1". 

Everything is working great. 

Hope it will help some guys!

**Comment by matks** [comment, marked as answer]
Happy to hear you found your answer 😉 

> I'm currently migrating from Prestashop 1.3 to the last version, 8.X.

Well 😅 you're trying to jump the software 10 years in the future.

It might be interesting to simply start a new shop in 8.X and migrate the data. I'm afraid almost nothing that works on PS 1.3 will work on PS 8.X . Modules, theme, customizations... there's such a big gap I'm afraid the only thing that can "travel" through time like this is data.

You'll find the SQL scripts that update the data structure in https://github.com/PrestaShop/autoupgrade/tree/dev/upgrade/sql and a documentation about the upgrade assistant here https://devdocs.prestashop-project.org/8/basics/keeping-up-to-date/upgrade-module/ . Be careful: these SQL files also contain links to PHP scripts ([example](https://github.com/PrestaShop/autoupgrade/blob/dev/upgrade/sql/1.1.0.1.sql#L2)) that need to be run using the Upgrade assistant.

**Reply by jdupont22** [reply]
It's exactly what I've done, have a fresh 8.X install and migrate the data through some personal SQL script, didn't know about this resource [](https://github.com/PrestaShop/autoupgrade/tree/dev/upgrade/sql)

Thanks @matks for the reply
